java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -248,6 +248,25 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_makeList(JNIEnv *env, j
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_makeListFromOffsets(
+    JNIEnv *env, jobject j_object, jlong child_handle, jlong offsets_handle, jlong row_count) {
+  JNI_NULL_CHECK(env, child_handle, "child_handle is null", 0)
+  JNI_NULL_CHECK(env, offsets_handle, "offsets_handle is null", 0)
+  try {
+    cudf::jni::auto_set_device(env);
+    auto const *child_cv = reinterpret_cast<cudf::column_view const *>(child_handle);
+    auto const *offsets_cv = reinterpret_cast<cudf::column_view const *>(offsets_handle);
+    CUDF_EXPECTS(offsets_cv->type().id() == cudf::type_id::INT32,
+                 "Input offsets does not have type INT32.");
+
+    auto result = cudf::make_lists_column(
+        static_cast<cudf::size_type>(row_count), std::make_unique<cudf::column>(*offsets_cv),
```

Comment:
This is not going to work.

From the docs for make_lists_column.

"The columns and mask are moved into the resulting lists column."

But the original column views passed in think that they own that memory.  You have to copy the data out at some point to make this valid.  We do this in other places in the plugin by creating a ColumnView from the corresponding information and then copying it out into a separate ColumnVector.  If you really want/need this API I would suggest that we implement it that way instead.

Reply:
No, that code above creates a new column by copying from the content of `offset_cv` so it's valid.

Reply:
So `std::make_unique<cudf::column>(*offsets_cv)` is copying the data, because you dereferenced the pointer?  This is for my info because I obviously didn't understand what was happening.

Reply:
Yes, `std::make_unique<cudf::column>` constructs a new (pointer containing) column. The `column` constructor taking a column view object (`*offsets_cv`) will copy data from that column into the new column.